index.js

```diff
@@ -3,71 +3,95 @@ function ocr(str) {
 }
 
 function scanDigit(str) {
-  const zero =  " _ " + 
-                "| |" + 
-                "|_|"
+  let digitsNonWhiteSpace = [
+    {'digit': 0, 'nonWhites': 6, 'uniques': [
+      {'index': 4, 'char': ' '}, 
+      {'index': 5, 'char': '|'}, 
+      {'index': 6, 'char': '|'}]},
+    {'digit': 1, 'nonWhites': 2, 'uniques': []},
+    {'digit': 2, 'nonWhites': 5, 'uniques': [
+      {'index': 3, 'char': ' '}, 
+      {'index': 5, 'char': '|'}, 
+      {'index': 6, 'char': '|'}, 
+      {'index': 8, 'char': ' '}]},
+    {'digit': 3, 'nonWhites': 5, 'uniques': [
+      {'index': 3, 'char': ' '}, 
+      {'index': 5, 'char': '|'}, 
+      {'index': 6, 'char': ' '}, 
+      {'index': 8, 'char': '|'}]},
+    {'digit': 4, 'nonWhites': 4, 'uniques': []},
+    {'digit': 5, 'nonWhites': 5, 'uniques': [
+      {'index': 3, 'char': '|'}, 
+      {'index': 5, 'char': ' '}, 
+      {'index': 6, 'char': ' '}, 
+      {'index': 8, 'char': '|'}]},
+    {'digit': 6, 'nonWhites': 6, 'uniques': [
+      {'index': 4, 'char': '_'}, 
+      {'index': 5, 'char': ' '}, 
+      {'index': 6, 'char': '|'}]},
+    {'digit': 7, 'nonWhites': 3, 'uniques': []},
+    {'digit': 8, 'nonWhites': 7, 'uniques': []},
+    {'digit': 9, 'nonWhites': 6, 'uniques': [
+      {'index': 4, 'char': '_'}, 
+      {'index': 5, 'char': '|'}, 
+      {'index': 6, 'char': ' '}]},
+  ]
 
-  const one =   "   " + 
-                "  |" + 
-                "  |"
-
-  const two =   " _ " + 
-                " _|" + 
-                "|_ "
-
-  const three = " _ " + 
-                " _|" + 
-                " _|"
-
-  const four =  "   " + 
-                "|_|" + 
-                "  |"
-
-  const five =  " _ " + 
-                "|_ " + 
-                " _|"
-
-  const six =   " _ " + 
-                "|_ " + 
-                "|_|"
-
-  const seven = " _ " + 
-                "  |" + 
-                "  |"
-
-  const eight = " _ " + 
-                "|_|" + 
-                "|_|"
-
-  const nine =  " _ " + 
-                "|_|" + 
-                " _|"
-
-  let digits = []
-  digits[zero] = 0
-  digits[one] = 1
-  digits[two] = 2
-  digits[three] = 3
-  digits[four] = 4
-  digits[five] = 5
-  digits[six] = 6
-  digits[seven] = 7
-  digits[eight] = 8
-  digits[nine] = 9
+  let nonWhites = (str.match(/[\|\_]/g)||[]).length
+  let possibleValues = digitsNonWhiteSpace.filter(function(digit) {
+    return digit.nonWhites === nonWhites
+  })
+  if (possibleValues.length === 1) {
+    return possibleValues[0].digit
+  }
+  let uniques = possibleValues.filter(function(possible) {
+    let perfectFit = true
```

Comment:
In future this can be a number to calculate the accuracy of a match: 1 or 0.85. It will be used to identify misread characters